### PR TITLE
Read from component

### PR DESCRIPTION
**Pull request by disrupted**

### CODE REVIEW

File: kpops/components/base_components/models/from_section.py [hunk 1]
```diff
@@ -19,24 +20,39 @@ class Config(BaseConfig):
...
+ComponentName = NewType("ComponentName", str)
+
+
 class FromSection(BaseModel):
-    topics: dict[str, FromTopic]
+    topics: dict[TopicName, FromTopic] = Field(
+        default={},
+        description="Topics to read from.",
+    )
+    components: dict[ComponentName, FromComponent] = Field(
```
Comment (philipp94831) [inline kpops/components/base_components/models/from_section.py:52]: Why aren't you using FromTopic here?

File: kpops/pipeline_generator/pipeline.py [hunk 2]
```diff
@@ -155,44 +145,44 @@ def parse_components(self, component_list: list[dict]) -> None:
...
-                    setattr(component.app, "name_override", component.name)
-
     def apply_component(
-        self,
-        component_data: dict,
-        component_class: type[PipelineComponent],
-        previous_component: PipelineComponent | None,
-    ) -> list[PipelineComponent]:
+        self, component_class: type[PipelineComponent], component_data: dict
+    ) -> None:
```
Comment (sujuka99) [inline kpops/pipeline_generator/pipeline.py:150]: I think this method would benefit from a docstring :)

File: kpops/components/base_components/models/from_section.py (hunk 1 above)
Comment (disrupted) [reply kpops/components/base_components/models/from_section.py:52]: you're right, I wasn't sure if I was gonna add anything else to it initially

**Review by philipp94831 — APPROVED** [review]

**Review by sujuka99 — APPROVED** [review]
LGTM

**Pull request merged**